### PR TITLE
snapstate: consider connect/disconnect tasks in CheckChangeConflict.

### PR DESCRIPTION
**Pull request by stolowski**
Inspect connect/disconnect tasks in CheckChangeConflict when looking for conflicting changes.
Note that multiple connect/disconnect shouldn't conflict with each other, as they are serialized by ifacemgr, thus there is exception for this case implemented via noConflictOnConnectTasks predicate.

### CODE REVIEW

File: overlord/ifacestate/ifacestate_test.go [hunk 1]
```diff
@@ -233,6 +233,39 @@ func (s *interfaceManagerSuite) TestDisconnectConflictsSlotSnap(c *C) {
 	s.testConnectDisconnectConflicts(c, ifacestate.Disconnect, "producer")
 }
 
+func (s *interfaceManagerSuite) TestConnectDoesntConflict(c *C) {
```
Comment (pedronis) [inline overlord/ifacestate/ifacestate_test.go]: I think we want a simple test also that if there's a connect/disconnect change,  snapstate.Update returns "snap has changes". See snapstate_test.go for examples of such tests

Comment (stolowski) [inline overlord/ifacestate/ifacestate_test.go]: Added.

File: overlord/snapstate/snapstate.go [hunk 2]
```diff
@@ -266,13 +267,27 @@ var snapTopicalTasks = map[string]bool{
...
+}
+
+func getPlugAndSlotRefs(task *state.Task) (interfaces.PlugRef, interfaces.SlotRef, error) {
+	var plugRef interfaces.PlugRef
+	var slotRef interfaces.SlotRef
+	if err := task.Get("plug", &plugRef); err != nil {
+		return plugRef, slotRef, err
+	}
+	if err := task.Get("slot", &slotRef); err != nil {
+		return plugRef, slotRef, err
```
Comment (chipaca) [inline overlord/snapstate/snapstate.go]: any reason you're returning the refs instead of `nil` here?

Comment (pedronis) [inline overlord/snapstate/snapstate.go]: this is returning structs, not pointer to structs, maybe that should indeed be changed though

Comment (pedronis) [inline overlord/snapstate/snapstate.go]: I pushed this change myself, let's see how it goes with the tests

Comment (stolowski) [inline overlord/snapstate/snapstate.go]: Thanks for pushing the change. NB, this helper function is a copycat of one from ifacestate/helpers.go, so perhaps a candidate for moving out in the future if we find a better common place for both.

Comment (pedronis) [inline overlord/snapstate/snapstate.go]: ah, interesting, the dup is a bit bad, not clear we have a good place to put this though

**Pull request merged**